ui/apps/platform/src/Components/CompoundSearchFilter/components/ConditionDate.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { Button, DatePicker, Flex, SelectOption } from '@patternfly/react-core';
+import { ArrowRightIcon } from '@patternfly/react-icons';
+
+import { getDate } from 'utils/dateUtils';
+import { dateConditions, ensureString } from '../utils/utils';
+
+import SimpleSelect from './SimpleSelect';
+
+export type ConditionDate = { condition: string; date: string };
+
+export type ConditionDateProps = {
+    value: ConditionDate;
+    onChange: (value: ConditionDate) => void;
+    onSearch: (value: ConditionDate) => void;
+};
+
+function dateParse(date: string): Date {
+    const split = date.split('/');
+    if (split.length !== 3) {
+        return new Date('Invalid Date');
+    }
+    const month = split[0];
+    const day = split[1];
+    const year = split[2];
+    if (month.length !== 2 || day.length !== 2 || year.length !== 4) {
+        return new Date('Invalid Date');
+    }
+    return new Date(
+        `${year.padStart(4, '0')}-${month.padStart(2, '0')}-${day.padStart(2, '0')}T00:00:00`
+    );
+}
+
+function ConditionDate({ value, onChange, onSearch }: ConditionDateProps) {
+    return (
+        <Flex spaceItems={{ default: 'spaceItemsNone' }}>
+            <SimpleSelect
+                value={value.condition}
+                onChange={(val) =>
+                    onChange({
+                        ...value,
+                        condition: ensureString(val),
+                    })
+                }
+                ariaLabelMenu="Condition selector menu"
+                ariaLabelToggle="Condition selector toggle"
+            >
+                {dateConditions.map((condition) => {
+                    return (
+                        <SelectOption key={condition} value={condition}>
+                            {condition}
+                        </SelectOption>
+                    );
+                })}
+            </SimpleSelect>
+            <DatePicker
+                aria-label="Filter by date"
+                buttonAriaLabel="Filter by date toggle"
+                value={value.date}
+                onChange={(_event, _value) => {
+                    onChange({ ...value, date: _value });
+                }}
```

Comment:
Nit: the underscore prefix on `_value` reads as if the value is unused at a quick glance.